frontend/src/components/Mobile/InputField.tsx

```diff
@@ -55,43 +61,44 @@ const StyledTitle = styled.div`
 `
 
 type InputFieldProps = {
-  title?: string
+  name: string
+  title: string
   placeholder: string
-  value?: string
-  setValue?: Dispatch<SetStateAction<string>> | ((input: string) => void)
   textArea?: boolean
-  hasError?: boolean
-  onChange?: () => void
+  required?: boolean
+  register
+  setValue
+  errors
 }
 
-export default function InputField({
-  title,
-  placeholder,
-  value,
-  setValue,
-  textArea,
-  onChange,
-  hasError,
-}: InputFieldProps) {
-  const [inputValue, setInputValue] = useState('')
+export default function InputField(props: InputFieldProps) {
+  const RedAsterisk = <RedText>*</RedText>
+  const { name, title, placeholder, textArea, required, register, setValue, errors } = props
 
   return (
     <Container>
-      {title && <StyledTitle>{title}</StyledTitle>}
+      {title && (
+        <StyledTitle>
+          {title}
+          {required && RedAsterisk}
+        </StyledTitle>
+      )}
       {textArea ? (
         <StyledTextArea
-          hasError={hasError}
-          placeholder={placeholder}
-          value={value ?? inputValue}
-          onChange={onChange ?? ((e) => (setValue ? setValue(e.target.value) : setInputValue(e.target.value)))}
+          placeholder={errors.name ? `${title} is required` : placeholder}
           rows={4}
+          onChange={(e) => setValue(name, e.target.value)}
+          hasError={!!errors.name}
```

Comment:
i think this will return the errors under the registered field 'name' instead of what you intended? not too sure